tests/system/robot/symbolPronunciationTests.py

```diff
@@ -536,20 +543,32 @@ def test_symbolInSpeechUI():
 	)
 
 
-def _setSymbolLevel(symbolLevel: SymLevel) -> None:
+def _setConfig(
+		symbolLevel: SymLevel,
```

Comment:
Now that this is more complex, can this get a sensible default too to match the default in configSpec, `SymLevel.SOME`

```suggestion
		symbolLevel: SymLevel: SymLevel.SOME,
```

Reply:
I am just wondering if this adds value here. Adding defaults for the 2 new properties was helpful because that way I didn't need to make as many changes. Every place that uses `_doTest` or `_setConfig` already specifies symbol level. Also although the default in `configSpec` for sym level is some, this value isn't actually used in any of these tests. So isn't adding a default for that param unnecessary and if it was to be added, wouldn't NONE or ALL be a more useful default value for these tests?

Reply:
I think it would be useful to have here in-case future tests do not wish to change the default value for symbol level, and instead wish to test the default symbol level.
Previously the function was `_setSymbolLevel` which was for explicitly changing the symbol from default. Now that it is `_setConfig` and empty call to `_setConfig` should not change the default config.

Reply:
I think that definitely makes sense. I am just concerned that this could be overengineering, since this is a change to support a testing pattern that isn't being used and might never be.

Consider that if I accepted this code change, the subsequent build would fail. Beyond the syntax being incorrect, SymLevel doesn't actually have a `SOME` member. It just has `NONE` and `ALL`, because that is the only options used in these tests. So if I was to make this change, I would also have to add that member. But since I would be adding it just in case someone in the future wanted to use it, then wouldn't it make sense to also add `MOST` in case someone wanted to use that in the future. And then it would make most sense to fill in the other enums like `IndentationReporting` with their missing options.

Instead wouldn't it be better if whoever would have such a use case in the future to be the one to add these things and make the changes that are necessary? I think that would avoid making changes and additions that might not even end up being used.

Reply:
I'd rather not accept a misleading function into the code base. I think the change of purpose to `_setConfig` warrants the change. The only additional required code involves adding a `SOME` member.

Reply:
But isn't it misleading to add the `SOME` member if it isn't actually being used in any test? I am not sure if I understand how `_setConfig` is now misleading. Back when it was `_setSymbolLevel` it didn't have any dfault and it wasn't considered misleading then. If the name of the function is what is misleading, then what if it was changed to something like `_changeConfig` or `_overrideConfig`?

@michaelDCurran @Qchristensen could we please get a third opinion on this thread?

Reply:
Alternatively, another option could be to return `_setSymbolLevel` to what it was and add a `_setIndentationReporting` function.

Reply:
Since we are dealing with configuration values here, I think they should either all be explicit, or all be optional with defaults.
As you don't wish to repeat indentation values that won't change much, I think the only option then is to make everything optional as @seanbudd  said.
However, I also like the alternative of adding _setIndentationReporting and keeping these functions separate.
So in short, if you are unwilling to make the symbol level value optional, then perhaps go with _setIndentationReporting.
However, as @seanbudd  is the NvDA core developer reviewing this pr, he should have the final say.

Reply:
@michaelDCurran thanks for your input

@seanbudd would you kindly make the code change you had in mind?

Reply:
That's been done